fluent/client/ws/connection.go

```diff
@@ -111,16 +118,21 @@ func (wsc *connection) CloseWithMsg(closeCode int, msg string) error {
 	return nil
 }
 
-func (wsc *connection) enqueueErr(err error) {
-	wsc.closedLock.Lock()
-	defer wsc.closedLock.Unlock()
+func (wsc *connection) Close() error {
+	return wsc.CloseWithMsg(websocket.CloseNormalClosure, "so long and thanks for all the fish")
```

Comment:
```
func (wsc *connection) CloseWithMsg(closeCode int, msg string) error {
	wsc.closedLock.Lock()
	defer wsc.closedLock.Unlock()

	if wsc.connClosed {
		return nil
	}

	wsc.connClosed = true

	err := wsc.WriteMessage(
		websocket.CloseMessage,
		websocket.FormatCloseMessage(
			closeCode, msg,
		),
	)

	if err != nil {
		log.Println("write close failed", err)
		return wsc.Conn.Close()
	}

	return nil
}
```

What happens when `wsc.WriteMessage` does not throw any error, what happens in this case?